### PR TITLE
[config] Improve Parse Error

### PR DESCRIPTION
**Pull request by davidiw**
Use a specific error type for parsing errors.

### CODE REVIEW

**Comment by bmwill** [comment]
I want to eventually move to using thiserror which would make implementing one-off error types like this much easier.

@bors-libra r+

File: config/src/config/mod.rs [hunk 1]
```diff
@@ -145,6 +146,21 @@ impl fmt::Display for RoleType {
...
+impl fmt::Display for ParseRoleError {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(fmt, "{}: {}", self.description(), &self.0)
+    }
+}
+
+impl std::error::Error for ParseRoleError {
+    fn description(&self) -> &str {
+        "Invalid node role"
+    }
```
Comment (dtolnay) [inline config/src/config/mod.rs]: The description method is deprecated and generally should not be implemented in new code.

Comment (bmwill) [reply config/src/config/mod.rs]: I have a set of commits which are converting everything to use anyhow and thiserror, i'll get rid of this there.

**Pull request closed without merging**